controllers/reconcilers/kserve_networkpolicy_reconciler.go

```diff
@@ -37,108 +34,24 @@ type KserveNetworkPolicyReconciler struct {
-		deltaProcessor:       processors.NewDeltaProcessor(),
 	}
 }
 
 func (r *KserveNetworkPolicyReconciler) Reconcile(ctx context.Context, log logr.Logger, isvc *kservev1beta1.InferenceService) error {
-
-	// Create Desired resource
-	desiredResource, err := r.createDesiredResource(isvc)
-	if err != nil {
-		return err
-	}
-
-	// Get Existing resource
-	existingResource, err := r.getExistingResource(ctx, log, isvc)
-	if err != nil {
-		return err
-	}
-
-	// Process Delta
-	if err = r.processDelta(ctx, log, desiredResource, existingResource); err != nil {
-		return err
-	}
-	return nil
-}
-
-func (r *KserveNetworkPolicyReconciler) createDesiredResource(isvc *kservev1beta1.InferenceService) (*v1.NetworkPolicy, error) {
-	desiredNetworkPolicy := &v1.NetworkPolicy{
-		ObjectMeta: metav1.ObjectMeta{
-			Name:      networkPolicyName,
-			Namespace: isvc.Namespace,
-			Labels: map[string]string{
-				"app.kubernetes.io/version":               "release-v1.9",
-				"networking.knative.dev/ingress-provider": "istio",
-			},
-		},
-		Spec: v1.NetworkPolicySpec{
-			PodSelector: metav1.LabelSelector{},
-			Ingress: []v1.NetworkPolicyIngressRule{
-				{
-					From: []v1.NetworkPolicyPeer{
-						{
-							NamespaceSelector: &metav1.LabelSelector{
-								MatchLabels: map[string]string{
-									"name": "openshift-user-workload-monitoring",
-								},
-							},
-						},
-					},
-				},
-			},
-			PolicyTypes: []v1.PolicyType{v1.PolicyTypeIngress},
-		},
-	}
-	return desiredNetworkPolicy, nil
+	return r.DeleteNetworkPolicy(ctx, isvc.Namespace)
 }
 
 func (r *KserveNetworkPolicyReconciler) getExistingResource(ctx context.Context, log logr.Logger, isvc *kservev1beta1.InferenceService) (*v1.NetworkPolicy, error) {
 	return r.networkPolicyHandler.FetchNetworkPolicy(ctx, log, types.NamespacedName{Name: networkPolicyName, Namespace: isvc.Namespace})
 }
 
-func (r *KserveNetworkPolicyReconciler) processDelta(ctx context.Context, log logr.Logger, desiredPod *v1.NetworkPolicy, existingPod *v1.NetworkPolicy) (err error) {
-	comparator := comparators.GetNetworkPolicyComparator()
-	delta := r.deltaProcessor.ComputeDelta(comparator, desiredPod, existingPod)
-
-	if !delta.HasChanges() {
-		log.V(1).Info("No delta found")
-		return nil
-	}
-
-	if delta.IsAdded() {
-		log.V(1).Info("Delta found", "create", desiredPod.GetName())
-		if err = r.client.Create(ctx, desiredPod); err != nil {
-			return
-		}
-	}
-	if delta.IsUpdated() {
-		log.V(1).Info("Delta found", "update", existingPod.GetName())
-		rp := existingPod.DeepCopy()
-		rp.Labels = desiredPod.Labels
-		rp.Spec = desiredPod.Spec
-
-		if err = r.client.Update(ctx, rp); err != nil {
-			return
-		}
-	}
-	if delta.IsRemoved() {
-		log.V(1).Info("Delta found", "delete", existingPod.GetName())
-		if err = r.client.Delete(ctx, existingPod); err != nil {
-			return
-		}
-	}
-	return nil
-}
-
 func (r *KserveNetworkPolicyReconciler) DeleteNetworkPolicy(ctx context.Context, isvcNamespace string) error {
```

Comment:
change the visiblity of `DeleteNetworkPolicy` function to private.

Reply:
Updated